crypto/rand/randfile.c

```diff
@@ -286,8 +286,22 @@ const char *RAND_file_name(char *buf, size_t size)
         if (OPENSSL_strlcpy(buf, s, size) >= size)
             return NULL;
     } else {
+#ifdef OPENSSL_SYS_WINDOWS
+        /*
+         * We use the same env variables as GetTempFile() - but that function
+         * uses TCHARs, but getenv() gives us chars so its easier to do it this
+         * way
```

Comment:
This is no excuse for doing it wrong. What's wrong? It won't work if there are non-ASCII letters in either of the variables, e.g. in username, which would be part of most variables in question. Because non-ASCII characters can be replaced with dummy ones, which will inevitably result in failure to open file. Well, doing right thing unfortunately harder than just modifying few lines in single function, but if we do something, it ought to be the right thing, not as wrong as it was. In this function right thing would be to call GetEnvironmentVariableW and convert result to UTF8. Then one has to ensure that RAND_load_file can process UTF8 file names in manner similar to file_fopen in bss_file.c.